File: tests/app/src/androidTestCoreDebug/java/com/example/cblverify/BasicCblInstrumentedTest.java
package com.example.cblverify;

import static org.junit.Assert.*;

import android.content.Context;
import android.util.Log;

import androidx.test.core.app.ApplicationProvider;
import androidx.test.ext.junit.runners.AndroidJUnit4;

import com.couchbase.lite.Blob;
import com.couchbase.lite.Collection;
import com.couchbase.lite.CouchbaseLite;
import com.couchbase.lite.CouchbaseLiteException;
import com.couchbase.lite.Database;
import com.couchbase.lite.DatabaseConfiguration;
import com.couchbase.lite.Document;
import com.couchbase.lite.MutableDocument;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.io.ByteArrayInputStream;
import java.nio.charset.StandardCharsets;

@RunWith(AndroidJUnit4.class)
public class BasicCblInstrumentedTest {

    private static final String DB_NAME = "release-verification";
    private Database db;

    @Before
    public void setUp() throws Exception {
        Context ctx = ApplicationProvider.getApplicationContext();
        CouchbaseLite.init(ctx);

        safeDeleteDb();

        DatabaseConfiguration config = new DatabaseConfiguration();
        db = new Database(DB_NAME, config);

        assertNotNull(db.getPath());
    }

    @After
    public void tearDown() {
        try {
            if (db != null) db.close();
        } catch (Exception ignore) {
        } finally {
            db = null;
        }

        safeDeleteDb();
    }

    private void safeDeleteDb() {
        try {
            if (Database.exists(DB_NAME, null)) {
                Database.delete(DB_NAME, null);
            }
        } catch (CouchbaseLiteException ignore) {
        }
    }

    @Test
    public void basicDatabaseAndBlobTest() throws Exception {
        Log.i("CBL-VERIFY", "Running flavor=" + BuildConfig.CBL_FLAVOR);
        assertEquals("core", BuildConfig.CBL_FLAVOR);
        Collection coll = db.getDefaultCollection();
        assertNotNull(coll);

        MutableDocument doc = new MutableDocument();
        doc.setString("firstname", "John");
        doc.setString("lastname", "Doe");

        byte[] data = "CouchbaseLite Database".getBytes(StandardCharsets.UTF_8);
        Blob blob = new Blob("text/plain", new ByteArrayInputStream(data));
        doc.setBlob("blob", blob);

        coll.save(doc);

        Document saved = coll.getDocument(doc.getId());
        assertNotNull(saved);
        assertEquals("John", saved.getString("firstname"));
        assertEquals("Doe", saved.getString("lastname"));
        assertNotNull(saved.getBlob("blob"));
    }
}

File: tests/app/src/androidTestKtxDebug/java/com/example/cblverify/BasicCblInstrumentedTest.kt
package com.example.cblverify

import android.content.Context
import android.util.Log
import androidx.test.core.app.ApplicationProvider
import androidx.test.ext.junit.runners.AndroidJUnit4
import com.couchbase.lite.*
import org.junit.After
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import java.io.ByteArrayInputStream

@RunWith(AndroidJUnit4::class)
class BasicCblKtxInstrumentedTest {

    private val dbName = "release-verification-ktx"
    private var db: Database? = null

    @Before
    fun setUp() {
        val ctx: Context = ApplicationProvider.getApplicationContext()
        CouchbaseLite.init(ctx)

        safeDeleteDb()
        db = Database(dbName, DatabaseConfiguration())
    }

    @After
    fun tearDown() {
        try { db?.close() } catch (_: Exception) {}
        db = null
        safeDeleteDb()
    }

    private fun safeDeleteDb() {
        try {
            if (Database.exists(dbName, null)) Database.delete(dbName, null)
        } catch (_: CouchbaseLiteException) {}
    }

    @Test
    fun basicDatabaseAndBlobTestKotlin() {
        Log.i("CBL-VERIFY", "Running flavor=" + BuildConfig.CBL_FLAVOR)
        assertEquals("ktx", BuildConfig.CBL_FLAVOR)
        val database = requireNotNull(db)

        val coll = database.defaultCollection
        val doc = MutableDocument()
            .setString("firstname", "John")
            .setString("lastname", "Doe")

        val data = "CouchbaseLite Database".encodeToByteArray()
        val blob = Blob("text/plain", ByteArrayInputStream(data))
        doc.setBlob("blob", blob)

        coll.save(doc)

        val saved = coll.getDocument(doc.id)
        requireNotNull(saved)
        assertEquals("John", saved.getString("firstname"))
        assertEquals("Doe", saved.getString("lastname"))
        requireNotNull(saved.getBlob("blob"))
    }
}
